pkg/agent/nodeportlocal/portcache/port_table_linux.go

```diff
@@ -183,66 +138,35 @@ func (pt *PortTable) AddRule(podIP string, podPort int, protocol string) (int, e
 			PodPort:   podPort,
 			Protocols: protocols,
 		}
-	}
-	protocolSocketData := npData.FindProtocol(protocol)
-	if protocolSocketData == nil {
-		return 0, fmt.Errorf("unknown protocol %s", protocol)
-	}
-	if protocolSocketData.State == stateInUse {
-		return 0, fmt.Errorf("rule for %s:%d:%s already exists", podIP, podPort, protocol)
-	}
-	if protocolSocketData.State == stateClosed {
-		return 0, fmt.Errorf("invalid socket state for %s:%d:%s", podIP, podPort, protocol)
-	}
-
-	nodePort := npData.NodePort
-	if err := pt.PodPortRules.AddRule(nodePort, podIP, podPort, protocol); err != nil {
-		return 0, err
-	}
-
-	protocolSocketData.State = stateInUse
-	if !exists {
-		pt.NodePortTable[strconv.Itoa(nodePort)] = npData
-		pt.PodEndpointTable[podIPPortFormat(podIP, podPort)] = npData
+		nodePort = npData.NodePort
+		if err := pt.PodPortRules.AddRule(nodePort, podIP, podPort, protocol); err != nil {
+			return 0, err
+		}
+		pt.NodePortTable[NodePortProtoFormat(nodePort, protocol)] = npData
```

Comment:
I'd like to use a cache.Indexer to store NodePortData, and provide to indexers: NodePort+prototol, and PodIP+PodPort+protocol, even PodIP as the third indexer. Then we don't need to main two maps, and to think about the data comsistence in them.

Reply:
Thanks, I will try cache.Indexer and see if it can work.

Reply:
Updated.